File: frontend/src/services/industry.ts
/**
 * Industry API Service
 */
import api from './api';

export interface IndustryJob {
  id: number;
  job_id: number;
  installer_id: number;
  facility_id: number;
  location_id: number;
  activity_id: number;
  blueprint_type_id: number;
  product_type_id: number | null;
  runs: number;
  cost: number | null;
  status: string;
  start_date: string;
  end_date: string;
  completed_date: string | null;
  successful_runs: number | null;
}

export interface IndustryFacility {
  id: number;
  facility_id: number;
  owner_id: number;
  solar_system_id: number;
  type_id: number | null;
  name: string | null;
  bonuses: Record<string, any> | null;
  tax: number | null;
}

export interface IndustryStatistics {
  total_jobs: number;
  active_jobs: number;
  completed_jobs: number;
  total_runs: number;
  total_cost: number;
  by_activity: Record<string, {
    jobs: number;
    runs: number;
    cost: number;
  }>;
}

export const industryService = {
  /**
   * List industry jobs
   */
  listJobs: async (params?: {
    character_id?: number;
    activity_id?: number;
    status?: string;
    limit?: number;
    offset?: number;
  }) => {
    const response = await api.get<IndustryJob[]>('/industry/jobs/', { params });
    return response.data;
  },

  /**
   * Get a specific industry job
   */
  getJob: async (jobId: number) => {
    const response = await api.get<IndustryJob>(`/industry/jobs/${jobId}`);
    return response.data;
  },

  /**
   * List industry facilities
   */
  listFacilities: async (params?: {
    character_id?: number;
    solar_system_id?: number;
    limit?: number;
    offset?: number;
  }) => {
    const response = await api.get<IndustryFacility[]>('/industry/facilities/', { params });
    return response.data;
  },

  /**
   * Get industry statistics
   */
  getStatistics: async (characterId: number, days: number = 30) => {
    const response = await api.get<IndustryStatistics>(`/industry/statistics/${characterId}`, {
      params: { days },
    });
    return response.data;
  },

  /**
   * Trigger industry sync
   */
  triggerSync: async (characterId: number) => {
    const response = await api.post(`/industry/sync/${characterId}`);
    return response.data;
  },
};

File: backend/app/models/industry.py
"""
Industry models for jobs and facilities
"""
from sqlalchemy import Column, Integer, BigInteger, String, Float, Boolean, DateTime, ForeignKey, Text, DECIMAL
from sqlalchemy.orm import relationship
from sqlalchemy.dialects.postgresql import JSONB
from datetime import datetime

from app.core.database import Base


class IndustryJob(Base):
    """Industry job model"""
    __tablename__ = "industry_jobs"

    id = Column(Integer, primary_key=True, index=True)
    character_id = Column(Integer, ForeignKey("characters.id"), nullable=False, index=True)
    job_id = Column(BigInteger, unique=True, nullable=False, index=True)

    # Job details
    installer_id = Column(BigInteger, nullable=False)
    facility_id = Column(BigInteger, nullable=False)
    location_id = Column(BigInteger, nullable=False)
    activity_id = Column(Integer, nullable=False)  # 1=manufacturing, 3=TE research, 4=ME research, 5=copying, 8=invention
    blueprint_id = Column(BigInteger, nullable=False)
    blueprint_type_id = Column(Integer, nullable=False)
    blueprint_location_id = Column(BigInteger, nullable=False)
    output_location_id = Column(BigInteger, nullable=False)

    # Product details
    product_type_id = Column(Integer)
    runs = Column(Integer, nullable=False)
    licensed_runs = Column(Integer)

    # Cost and probability
    cost = Column(DECIMAL(precision=20, scale=2))
    probability = Column(Float)

    # Timing
    status = Column(String(50), nullable=False)  # active, paused, ready, delivered, cancelled, reverted
    start_date = Column(DateTime(timezone=True), nullable=False, index=True)
    end_date = Column(DateTime(timezone=True), nullable=False, index=True)
    pause_date = Column(DateTime(timezone=True))
    completed_date = Column(DateTime(timezone=True))

    # Success metrics
    completed_character_id = Column(BigInteger)
    successful_runs = Column(Integer)

    # Metadata
    created_at = Column(DateTime(timezone=True), default=datetime.utcnow)
    updated_at = Column(DateTime(timezone=True), default=datetime.utcnow, onupdate=datetime.utcnow)

    # Relationships
    character = relationship("Character", back_populates="industry_jobs")


class IndustryFacility(Base):
    """Industry facility/structure model"""
    __tablename__ = "industry_facilities"

    id = Column(Integer, primary_key=True, index=True)
    character_id = Column(Integer, ForeignKey("characters.id"), nullable=False, index=True)
    facility_id = Column(BigInteger, unique=True, nullable=False, index=True)

    # Facility details
    owner_id = Column(BigInteger, nullable=False)
    solar_system_id = Column(BigInteger, nullable=False, index=True)
    type_id = Column(Integer)
    name = Column(String(255))

    # Bonuses (JSONB for flexible activity bonuses)
    # Format: {"manufacturing": {"material": 0.01, "time": 0.25}, "research_time": {"material": 0.0, "time": 0.25}}
    bonuses = Column(JSONB)

    # Tax rates
    tax = Column(Float)

    # Metadata
    created_at = Column(DateTime(timezone=True), default=datetime.utcnow)
    updated_at = Column(DateTime(timezone=True), default=datetime.utcnow, onupdate=datetime.utcnow)

    # Relationships
    character = relationship("Character", back_populates="industry_facilities")


class IndustryActivity(Base):
    """Industry activity tracking for analytics"""
    __tablename__ = "industry_activities"

    id = Column(Integer, primary_key=True, index=True)
    character_id = Column(Integer, ForeignKey("characters.id"), nullable=False, index=True)

    # Activity summary
    activity_id = Column(Integer, nullable=False, index=True)
    activity_name = Column(String(100))

    # Metrics
    total_jobs = Column(Integer, default=0)
    completed_jobs = Column(Integer, default=0)
    active_jobs = Column(Integer, default=0)
    total_runs = Column(Integer, default=0)
    total_cost = Column(DECIMAL(precision=20, scale=2), default=0)

    # Time period
    period_start = Column(DateTime(timezone=True), nullable=False, index=True)
    period_end = Column(DateTime(timezone=True), nullable=False)

    # Metadata
    created_at = Column(DateTime(timezone=True), default=datetime.utcnow)
    updated_at = Column(DateTime(timezone=True), default=datetime.utcnow, onupdate=datetime.utcnow)

    # Relationships
    character = relationship("Character", back_populates="industry_activities")
